src/bun.js/bindings/bindings.cpp

```diff
@@ -3027,6 +3027,18 @@ void JSC__JSValue__put(JSC__JSValue JSValue0, JSC__JSGlobalObject* arg1, const Z
     object->putDirect(arg1->vm(), Zig::toIdentifier(*arg2, arg1), JSC::JSValue::decode(JSValue3));
 }
 
+extern "C" void JSC__JSValue__putMayBeIndex(JSC__JSValue target, JSC__JSGlobalObject* globalObject, const BunString* key, JSC__JSValue value)
+{
+    JSC::VM& vm = globalObject->vm();
+
+    WTF::String keyStr = key->toWTFString();
+    JSC::Identifier identifier = JSC::Identifier::fromString(vm, keyStr);
+
+    JSC::JSObject* object = JSC::JSValue::decode(target).asCell()->getObject();
+    object->putDirectMayBeIndex(globalObject, JSC::PropertyName(identifier), JSC::JSValue::decode(value));
```

Comment:
this could throw if the object is a proxy object. can you add exception checking?

Reply:
You mean a `DECLARE_THROW_SCOPE` / `RETURN_IF_EXCEPTION`? Or something else?
Sorry, I'm not really sure how exception handling works

Reply:
Yep, `DECLARE_THROW_SCOPE` and `RETURN_IF_EXCEPTION` with the scope and the return value. You can find examples of this pattern in `bindings.cpp` if you search for `getIfPropertyExists`